src/Dapr.Client/DaprClient.cs

```diff
@@ -756,6 +756,44 @@ public async Task<StateEntry<TValue>> GetStateEntryAsync<TValue>(string storeNam
             IReadOnlyDictionary<string, string> metadata = default,
             CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Queries a statestore with the given request. The request contains a JSON representation of the query as described by the Dapr QueryState API.
+        /// Note that the underlying statestore must support queries.
+        /// </summary>
+        /// <param name="request">A <see cref="StateQueryRequest"/></param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <typeparam name="TValue">The data type of the value to read.</typeparam>
+        /// <returns>A <see cref="StateQueryResponse{TValue}"/> that may be paginated, use <see cref="StateQueryResponse{TValue}.Token"/> to continue the query.</returns>
+        public abstract Task<StateQueryResponse<TValue>> QueryStateAsync<TValue>(
+            StateQueryRequest request,
```

Comment:
Should metadata be its own parameter here?

Reply:
After looking at this in more detail, it seems best to just get rid of this overload and have a single one. It's just 3 parameters and one of them is optional anyway.